File: client-java/controller-api/src/main/java/org/evomaster/client/java/controller/api/dto/database/execution/epa/RestActions.java
package org.evomaster.client.java.controller.api.dto.database.execution.epa;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.HashSet;

public class RestActions {
    @JsonProperty("enabledRestActions")
    public HashSet<RestAction> enabledRestActions;

    public RestActions() {
        this.enabledRestActions = new HashSet<>();
    }

    public RestActions(@JsonProperty("enabledRestActions") HashSet<RestAction> enabledRestActions) {
        this.enabledRestActions = enabledRestActions;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        for (RestAction a : enabledRestActions) {
            sb.append(String.format("%s\n", a.toString()));
        }
        return sb.toString();
    }

}

File: core/src/main/kotlin/org/evomaster/core/problem/rest/epa/RestActions.kt
package org.evomaster.client.java.controller.api.dto.database.execution.epa

import kotlin.collections.HashSet

class RestActions (
    var enabledRestActions: HashSet<RestAction> = HashSet()
) {
    fun toStringForEPA(): String {
        return enabledRestActions.stream()
            .map { obj: RestAction -> obj.toString() }
            .sorted()
            .reduce { s: String?, a: String? -> "$s, \\n$a" }
            .orElse("")
    }

    companion object {
        fun from(restActionsDto: RestActionsDto) : RestActions {
            val set = restActionsDto.enabledRestActions.stream().map { RestAction(it.verb, it.path) }.toArray().toHashSet() as HashSet<RestAction>
            return RestActions(set)
        }
    }
}